### DISCUSSION TITLE
Confusion/Error around Neorg directory struture and linking

### DISCUSSION THREAD

**Discussion by AndOrangutan** [nvim-neorg/neorg#147] [Q&A] [answered]
Sorry, that I am back again.

I am either confused or having an error in regard to how to set up a directory structure or links. I have spent the past 2 hours or so really trying to figure this out (reading through the README and docs on this GitHub page), and am really stuck, but you still have my deepest apologies if this is somehow just something I failed to research.

So here is my directory structure for my Neorg (which I keep in my Dropbox folder to keep backed up).
![NeorgTree](https://user-images.githubusercontent.com/40435989/136475369-bd39f537-bcbc-4768-9718-b423d0dacf99.png)
The directory I am focusing on right now is the phil103 one.

The link to my `| PhilTest` marker in `/book_o_everything/book_o_phil103/politics_and_speech_on_the_internet.norg` works from the head `index.norg` as one would expect. Great :D. 
![NeorgWorkLink](https://user-images.githubusercontent.com/40435989/136476521-cc7ed89d-f482-462d-8e63-8d9f257cbe4f.png)

Now for the fun that is jumping into my  `/book_o_everything/book_o_phil103/book_o_phil103.norg`. First I tried just writing my own link (which I now think I did incorrectly after seeing the auto-generated link, which i will get to next.). You can see I get this error.
![NeorgError1](https://user-images.githubusercontent.com/40435989/136476066-8f4183d0-cc77-4e2c-b39d-1c866fe6503a.png)

I figured this was just because I am bad (which I definitely am), so I decided to use the telescope extension link generator thingy, and it spat out this. 
![NeorgError2](https://user-images.githubusercontent.com/40435989/136476258-64961ea6-aee4-4c7f-a3c6-1afdf91c1bc8.png)
This link didn't work either, despite being the exact same link that works in the header. 

Am I missing something?

Edit: Added pic of head index.norg (which works)

**Comment by vhyrro** [comment, marked as answer]
Ah, you should be getting a popup telling you the link is invalid, we recently overhauled the selection popup UI and it seems the old code snuck in somewhere. As for exacly why your link doesn't work it seems to be because of relative vs absolute paths. If you're in book_o_phil103.norg then you can simply do `[test](:politics_and_speech_on_the_internet:|PhilTest)`, since that file is in the same directory as the former norg file. The telescope extension seems to trip a bit and always place relative paths, which it shouldn't. *If you try:* `[test](:/book_o_everything/book_o_phil103/politics_and_speech_on_the_internet:|PhilTest)` (note the / at the beginning which tells neorg to start at the root of the workspace) it should hopefully work. Whenever you get that "create_selection not found" error then just assume that the link has valid syntax but its destination couldn't be found. I'll be fixing that today :)

**Reply by AndOrangutan** [reply]
Okay, that fixed it, thanks for the help :D

**Comment by vhyrro** [comment]
also dont worry about making a lot of issues/github discussions, it's actually really important to us to understand where people are tripping up the most and where we gotta fix bugs/improve!
